source/bayernfahrplan/fahrplanparser/json/jsonUtils.d

```diff
@@ -0,0 +1,222 @@
+        classUnderTest.getDepartureTime.should.equal(DateTime(2018, 1, 1, 0, 1, 0));
+    }
+
+    unittest
+    {
+        import std.json : JSONException;
+
+        auto classUnderTest = JSONValue("");
+        classUnderTest.getDepartureTime.should.throwException!JSONException;
+    }
+
+    unittest
+    {
+        import std.json : JSONException;
+
+        auto classUnderTest = JSONValue(["dateTime" : ["time" : "00:01"]]);
+        classUnderTest.getDepartureTime.should.throwException!JSONException;
+    }
+
+    unittest
+    {
+        import std.json : JSONException;
+
+        auto classUnderTest = JSONValue(["dateTime" : ["date" : "01.01.2018"]]);
+        classUnderTest.getDepartureTime.should.throwException!JSONException;
+    }
+
+    unittest
+    {
+        import std.json : JSONException;
+
+        auto classUnderTest = JSONValue(["dateTime" : ["date" : "01.01.2018", "time" : "00:01"]]);
+        classUnderTest.getDepartureTime.should.equal(DateTime(2018, 1, 1, 0, 1, 0));
+    }
+}
+
+DateTime getRealDepartureTime(ref in JSONValue departureInfo)
+{
+    // TODO : Error handling
+    if (departureInfo["realtime"].integer == 1)
+    {
+        // TODO: Extract
+        return DateTime(departureInfo["dateTime"]["rtDate"].str.parseDefasDate,
+                TimeOfDay.fromISOExtString(departureInfo["dateTime"]["rtTime"].str ~ ":00"));
+    }
+    else
+    {
+        return departureInfo.getDepartureTime;
+    }
+}
+
+@system
+{
+    import std.json : parseJSON;
+
+    unittest
+    {
+        auto classUnderTest = `{"realtime": 1, "dateTime": {"rtDate": "1.1.2018", "rtTime": "00:01"}}`
+            .parseJSON;
+        classUnderTest.getRealDepartureTime.should.equal(DateTime(2018, 1, 1, 00, 01, 00));
+    }
+
+    unittest
+    {
+        import std.json : JSONException;
+
+        auto classUnderTest = `{"realtime": 1, "dateTime": {"rtDate": "1.1.2018"}}`.parseJSON;
+        classUnderTest.getRealDepartureTime.should.throwException!JSONException;
+    }
+
+    unittest
+    {
+        import std.json : JSONException;
+
+        auto classUnderTest = `{"realtime": 1, "dateTime": {"rtTime": "00:01"}}`.parseJSON;
+        classUnderTest.getRealDepartureTime.should.throwException!JSONException;
+    }
+
+    unittest
+    {
+        import std.json : JSONException;
+
+        auto classUnderTest = `{"dateTime": {"rtDate": "1.1.2018", "rtTime": "00:01"}}`.parseJSON;
+        classUnderTest.getRealDepartureTime.should.throwException!JSONException;
+    }
+
+    unittest
+    {
+        auto classUnderTest = `{"realtime": 0,
+                "dateTime": {"date": "1.1.2018", "time": "00:01", "rtDate": "1.1.2018", "rtTime": "00:05"}}`
+            .parseJSON;
+        classUnderTest.getRealDepartureTime.should.not.throwAnyException;
+        classUnderTest.getRealDepartureTime.should.equal(DateTime(2018, 1, 1, 00, 01, 00));
+    }
+}
+
+Date parseDefasDate(string dateString)
```

Comment:
If we use `departureInfo["stamp"][...]` instead of `departureInfo["dateTime"][...]`, we can simplify this: Then it is only a call to `Date.fromISOString()`.

For consistency, we should also use `"stamp"` for parsing the time.

`dateString` can be `const`.